fetch.bs

```diff
@@ -5630,6 +5622,82 @@ if the script checks that the URL has the right hostname.
 
 
 
+<h2 id=data-urls><code>data:</code> URLs</h2>
+
+<p>For an informative description of <code>data:</code> URLs, see RFC 2397. This section replaces
+that RFC's normative processing requirements. [[RFC2397]]
+
+<p>The <dfn export><code>data:</code> URL processor</dfn> takes a <a for=/>URL</a>
+<var>dataURL</var> and then runs these steps:
+
+<ol>
+ <li><p>Assert: <var>dataURL</var>'s <a for=URL>scheme</a> is "<code>data</code>".
+
+ <li><p>Let <var>input</var> be the result of running the <a>URL serializer</a> on
+ <var>dataURL</var> with the <i>exclude fragment flag</i> set.
+
+ <li><p>Remove the leading "<code>data:</code>" string from <var>input</var>.
+
+ <li><p>Let <var>position</var> point at the start of <var>input</var>.
+
+ <li><p>Let <var>encodedMimeType</var> be the result of
+ <a>collecting a sequence of code points</a> that are not equal to U+002C (,), given
+ <var>position</var>.
+
+ <li><p>If <var>position</var> is past the end of <var>input</var>, then return failure.
+
+ <li><p>Advance <var>position</var> by 1.
+
+ <li><p>Let <var>encodedBody</var> be the remainder of <var>input</var>.
+
+ <li><p>Let <var>mimeTypeBytes</var> be the <a>string percent decoding</a> of
+ <var>encodedMimeType</var>.
+ <!-- Note: implementations leave the percent-encoded bits around. That strikes me as rather broken,
+      but it might be due to web compatibility? -->
+
+ <li><p>Let <var>mimeType</var> be the <a>isomorphic decode</a> of <var>mimeTypeBytes</var>.
+
+ <li><p><a>Strip leading and trailing ASCII whitespace</a> from <var>mimeType</var>.
+
+ <li><p>Let <var>body</var> be the <a>string percent decoding</a> of <var>encodedBody</var>.
+
+ <li>
+  <p>If <var>mimeType</var> contains an <a>ASCII case-insensitive</a> match for
+  "<code>;base64;</code>" or ends with an <a>ASCII case-insensitive</a> match for
```

Comment:
Is no whitespace around "base64" allowed? The URL `data:text/html ; base64 ;,YW5uZXZr` works for me on ChromeOS, so tests will be needed for this.

Reply:
The silly thing is that `data:text/html%20;%20base64%20;,YW5uZXZr` doesn't work, but I also cannot get that to work in other browsers (tested Firefox and Safari).

Reply:
Is that not just because of the above issue about percent decoding the media type?

When it comes to whitespace, is there an ideal behavior that we should try to do, or should we just look for what's most web compatible?

Reply:
I think ideal might relate to how we parse MIME types in general. Ignoring whitespace is what we typically do though so I suppose that makes sense, but will require more of an actual parser.